hetmech/degree_weight.py

```diff
@@ -30,25 +30,29 @@ def dwwc_step(
     numpy.ndarray
         Normalized matrix with dtype.float64.
     """
-    # returns a newly allocated numpy.ndarray
-    matrix = numpy.array(matrix, numpy.float64, copy=copy)
+    # returns a newly allocated array
+    mat_type = type(matrix)
+    if mat_type == numpy.ndarray:
+        mat_type = numpy.array
+    matrix = mat_type(matrix, dtype=numpy.float64, copy=copy)
     assert matrix.ndim == 2
 
-    row_sums = matrix.sum(axis=1)
-    column_sums = matrix.sum(axis=0)
+    row_sums = numpy.array(matrix.sum(axis=1)).flatten()
```

Comment:
I put that in there because, as far as I can tell, giving `matrix.sum(axis=` type sums gives the same results for sparse and dense. The only difference is that for sums over sparses, the result is given as a list of lists instead of a simple list. The elements are in the right order, I just use `flatten` to make the output identical to `numpy.array` sums. The line in question does not change a dense array's output, and it just gives the same output as it did before.